### ISSUE TITLE
"St" for text-to-speech

### ISSUE THREAD

**Issue by mvl22** [cyclestreets/android#442]
User feedback:

The Android app seems to be expanding "St" incorrectly with text to speech. I don't think it did this before the changes to speaking road numbers (it, or Android, used to have trouble with "St." with a dot). It was expanding it to "Street" when I was cycling up [St Cross Road in Winchester](https://www.openstreetmap.org/way/100098754#map=16/51.0475/-1.3181&layers=CN). And yes - the signs do have "St" rather than "Saint".

**Comment by mvl22** [comment]
Presumably we need to treat "St" as "Street" if it's at the end, and as "Saint" if it's at the start.

**Comment by oliverlockwood** [comment]
@mvl22 do you have an example route you could share which exhibits this issue, please?

**Comment by mvl22** [comment]
I assume that this example, which uses the road that the original feedbacker gave, would exhibit the problem:

https://www.cyclestreets.net/journey/71473516/

**Comment by oliverlockwood** [comment]
Interesting.

I just created https://cycle.st/j71474135 - a journey that ends on the road in question, as then I could just start up LiveRide on my phone and hear the starting audio.

It correctly pronounced "Saint Cross Road".

So I suspect this may be something in the particular phone's TTS engine - we have seen variations like this before.

That being said, I'll see if we can put in some magic around this, in the same way that we did to pronounce numerical roads more clearly.
